lib/awful/hotkeys_popup/widget.lua

```diff
@@ -132,11 +138,24 @@ local function add_hotkey(key, data, target)
     local group = data.group or "none"
     group_list[group] = true
     if not target[group] then target[group] = {} end
-    table.insert(
-        target[group],
-        {hotkey= joined_mods .. (widget.labels[key] or key),
-         description=data.description}
-    )
+    local new_key = {
+        key = (widget.labels[key] or key),
+        mod = joined_mods,
+        description = data.description
+    }
+    local index = data.description or "none"  -- or use its hash?
```

Comment:
Last time we (@blueyed or me) tried to use GLib string hashing API, we got a nice LGI segfault. So this will do for now. I don't see issues leaving the comment, at some point it would indeed be nice to use hashing.